### DISCUSSION TITLE
SortOrder vs SortOrderInput

### DISCUSSION THREAD

**Discussion by ericwink** [MichalLytek/typegraphql-prisma#430] [General]
Hi folks,

I'm upgrading from older dependencies, and now running current Prisma Typegraphql I'm seeing a discrepancy in how OrderBy is handled when compared to previous versions.

It seems that if something is marked as required in my Prisma Model, the generated OrderBy file assigns the following:

```
  @TypeGraphQL.Field(_type => SortOrder, {
    nullable: true
  })
  name?: "asc" | "desc" | undefined;
```
  
 However, if something is marked as optional in my Prisma schema, it results in the following:
 
```
  @TypeGraphQL.Field(_type => SortOrderInput, {
    nullable: true
  })
  name?: SortOrderInput | undefined;
```
  
SortOrderInput is defined as...

```
export class SortOrderInput {
  @TypeGraphQL.Field(_type => SortOrder, {
    nullable: false
  })
  sort!: "asc" | "desc";

  @TypeGraphQL.Field(_type => NullsOrder, {
    nullable: true
  })
  nulls?: "first" | "last" | undefined;
}
```

This is challenging, as when trying to allow dynamic sorting controlled by the user, I don't know how to determine when I can just pass `orderBy: [{key: 'asc'}]` or when I need to pass `orderBy: [{key: {sort: 'asc'}}]`

Is there any way to make everything follow just the SortOrder logic?

Thank you!

**Comment by MichalLytek** [comment]
The logic is as you see - for nullable columns you can sort by nulls. There's no abstraction to unify this.

**Reply by ericwink** [reply]
Thanks so much for the quick reply! Knowing my options helps me focus on the path forward so this was definitely helpful. And thank you for all of your great work!

**Reply by chriszrc** [reply]
How are other people working around this? I literally created a lookup object of all the column names for a table, and have the string representation of the type and have that object use a type annotation to validate that the the lookup object still conforms to the database table type, all so I can just set the right format for the SortOrder. I realize the types are just following Prisma here, but there must be a better way to handle this when dynamically generating the grapqhl order value?
